### ISSUE TITLE
Pooch update breaks import

### ISSUE THREAD

**Issue by manthey** [scikit-image/scikit-image#5528]
## Description

Pooch recently updated to version 1.5.0.  This moved the hash functions to their own module, so the current method of importing them now fails.  Further, the fallback for if pooch is not installed doesn't work because the exact error is different.

## Way to reproduce
```python
import skimage.metrics
```
Note, install pooch 1.5.0 first.

You'll see
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../python3.8/site-packages/skimage/__init__.py", line 135, in <module>
    from .data import data_dir
  File ".../python3.8/site-packages/skimage/data/__init__.py", line 64, in <module>
    from pooch.utils import file_hash
ImportError: cannot import name 'file_hash' from 'pooch.utils' (.../python3.8/site-packages/pooch/utils.py)
```

## Version information
```python
# Paste the output of the following python commands
from __future__ import print_function
import sys; print(sys.version)
import platform; print(platform.platform())
import skimage; print("scikit-image version: {}".format(skimage.__version__))
import numpy; print("numpy version: {}".format(numpy.__version__))
```

```python
# your output here
3.8.10 (default, May  5 2021, 03:01:07)
[GCC 7.5.0]
Linux-5.4.0-80-generic-x86_64-with-glibc2.27
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../python3.8/site-packages/skimage/__init__.py", line 135, in <module>
    from .data import data_dir
  File ".../python3.8/site-packages/skimage/data/__init__.py", line 64, in <module>
    from pooch.utils import file_hash
ImportError: cannot import name 'file_hash' from 'pooch.utils' (.../python3.8/site-packages/pooch/utils.py)
numpy version: 1.21.2
```

I'm using ```scikit-image version: 0.18.2```

## Possible Fix

One solution would be to change `skimage/data/__init__.py`, where it reads
```
    from pooch.utils import file_hash
```
to
```
    try:
        from pooch.utils import file_hash
    except ImportError:
        from pooch.hashes import file_hash
```

**Comment by hmaarrfk** [comment]
ah thanks for this. I had actually asked for that pooch release myself... thanks for flagging this.

**Comment by hmaarrfk** [comment]
I'm also asking to see if pooch wants to fix this themselves:

https://github.com/fatiando/pooch/pull/251

**Comment by hmaarrfk** [comment]
The issue is that I think that pooch does not consider this to be in their API. In our initial PR getting this pooch functionality in, we decided to import their function instead of copying the 5 lines of code.

I think that maybe we should copy the code.

**Comment by leouieda** [comment]
@hmaarrfk the `file_hash` function is a part of the API but the modules are not (see https://www.fatiando.org/pooch/latest/api/generated/pooch.file_hash.html#pooch.file_hash). Notice that everything in the docs use accessed through the top level `pooch` namespace. 

We can add the backwards compatible fix for now if that makes things easier for everyone and remove it in the next release. 

Sorry for all the confusion.

**Comment by hmaarrfk** [comment]
We must have crossed messages. grlee did find that as well and we changed the import statements to match. Now onto releasing software without the bug!

**Comment by leouieda** [comment]
@hmaarrfk just saw the PR fixing this. Again, sorry for the confusion. I guess we assumed that the "public API" was what was defined in the docs but never really spelled it out like that. Definitely a learning moment for me.

**Comment by grlee77** [comment]
> We can add the backwards compatible fix for now if that makes things easier for everyone and remove it in the next release.

This would be nice, especially as it it looks `napari` also encountered the same issue. We have resolved it on main here now, but it may take us a bit longer to go through the release process.

>  Again, sorry for the confusion. I guess we assumed that the "public API" was what was defined in the docs but never really spelled it out like that. Definitely a learning moment for me.

no worries. Thanks for the quick feedback and for creating a useful library!

**Comment by leouieda** [comment]
Alright, I'll cut a new release today (just waiting on another PR with a fix to our tests). Thanks for the quick feedback and patience 🙂

**Comment by leouieda** [comment]
Pooch v1.5.1 is out on PyPI and soon on conda-forge as well. It allows importing `file_hash` from `pooch.utils` but issues a deprecation warning to import from the top-level namespace instead. This is backwards compatible all the way back to our first release so together with #5529 it should fix things for the foreseeable future.

**Comment by grlee77** [comment]
Thanks @leouieda, we are in the process of getting a new release out here as well (v0.18.3 wheel builds are still running)
